x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/results/add_result_logic.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../../../shared/flash_messages';
+import { HttpLogic } from '../../../../../shared/http';
+
+import { EngineLogic } from '../../../engine';
+import { Result } from '../../../result/types';
+
+interface AddResultValues {
+  isFlyoutOpen: boolean;
+  dataLoading: boolean;
+  searchQuery: string;
+  searchResults: Result[];
+}
+
+interface AddResultActions {
+  openFlyout(): void;
+  closeFlyout(): void;
+  search(query: string): { query: string };
+  onSearch({ results }: { results: Result[] }): { results: Result[] };
+}
+
+export const AddResultLogic = kea<MakeLogicType<AddResultValues, AddResultActions>>({
+  path: ['enterprise_search', 'app_search', 'curation_add_result_logic'],
+  actions: () => ({
+    openFlyout: true,
+    closeFlyout: true,
+    search: (query) => ({ query }),
+    onSearch: ({ results }) => ({ results }),
+  }),
+  reducers: () => ({
+    isFlyoutOpen: [
+      false,
+      {
+        openFlyout: () => true,
+        closeFlyout: () => false,
+      },
+    ],
+    dataLoading: [
+      false,
+      {
+        search: () => true,
+        onSearch: () => false,
+      },
+    ],
+    searchQuery: [
+      '',
+      {
+        search: (_, { query }) => query,
```

Comment:
Oh yeah, one more question for y'all!! I was wondering about this when I was recording the screen capture for the PR, but should I be resetting the search query back to `''` whenever the flyout opens/closes? 🤔 Was it weird that the search input persisted...? Or was it _cool?_... maaybe??...

Reply:
👍 to clearing state on modal open/close!

> I guess because the AddResultLogic state is higher in the component tree than AddResultFlyout, it never unmounts

FWIW even if we removed `{isFlyoutOpen && <AddResultFlyout />}` from Curation.tsx, state would still be preserved on close. This is because we're calling `const { flyoutOpen } = useActions(AddResultLogic)` in PromotedDocuments & HiddenDocuments, and `AddResultLogic` being present in those views preserves the logic file and keeps it from unmounting. 🤷 

I think for that reason I prefer keeping the conditionality of the flyout on the top-level view (so it's super clear that the flyout is always mounted).

Reply:
Commit that resets the search query: https://github.com/elastic/kibana/pull/94887/commits/6c559ac9a37bac03d523440d8fe20be72ce7511f